abdiff/core/build_ab_images.py

```diff
@@ -1 +1,67 @@
-"""abdiff.core.build_ab_images"""
+import logging
+import os
+import shutil
+
+import docker
+from pygit2 import clone_repository
+from pygit2.enums import ResetMode
+
+from abdiff.core.utils import update_or_create_job_json
+
+logger = logging.getLogger(__name__)
+
+
+def build_ab_images(
+    job_directory: str,
+    docker_client: docker.client.DockerClient,
+    commit_sha_a: str,
+    commit_sha_b: str,
+) -> tuple[str, str]:
+    """Build Docker images based on 2 commit SHAs.
+
+    Args:
+        job_directory: The directory containing all files related to a job.
+        docker_client: A configured Docker client.
+        commit_sha_a: The SHA of the first commit for comparison.
+        commit_sha_b: The SHA of the second commit for comparison.
+    """
+    images = []
+    for commit_sha in [commit_sha_a, commit_sha_b]:
+        clone_directory = f"{job_directory}/{commit_sha}"
```

Comment:
If I'm understand this correctly, this is temporarily cloning transmogrifier into the job directory under a particular commit SHA?

I wonder if this is a good use for [python's `tempfile` module](https://docs.python.org/3/library/tempfile.html)?  This provides a temporary location on disk to do things, but is _automatically_ cleaned up after the python process closes. Example of a getting a temp dir:

```python
with tempfile.TemporaryDirectory() as tmpdir:
    print(tmpdir)
# /var/folders/26/b99kw7554l7bgrx51srd_7tr0000gp/T/tmpr7gdefsg
```

In this way, it's possible you may not need the method `delete_repo_clone()` at all, as we could rely on `tempfile` to automatically cleanup when the python process completes.

Reply:
This is great and was simple to implement, great suggestion!